### ISSUE TITLE
URI params method

### ISSUE THREAD

**Issue by vitaliihonta** [pepegar/hammock#98]
Hi there!
Could you make a `param(s)` method of `Uri` class like in `STTP` library? It would be very useful.
Here is an implicit class I'm using in my production code:

```scala
  implicit class UriOps(val self: Uri) extends AnyVal {
    def param(key: String, value: String): Uri = self.copy(query = self.query + (key → value))
    def params(ps: (String, String)*): Uri = ps match {
      case Seq() ⇒ self
      case _     ⇒ ps.foldLeft(self) { case (uri, (k, v)) ⇒ uri.copy(query = uri.query + (k → v)) }
    }
  }
```

I could extend Uri class or provide this implicit class by myself and open a pull request

**Comment by pepegar** [comment]
Hi @vitaliihonta !

Sure, it's a great idea in fact. I've been wanting to extend the functionality on the Uri datatype for some time.

I think that for now is better to extend the `Uri` class itself, what do you think?

**Comment by pepegar** [comment]
Do you want to open the PR for it?

**Comment by vitaliihonta** [comment]
@pepegar great, so I'll start implementing it in a few minutes

**Comment by pepegar** [comment]
Awesome, thanks @vitaliihonta !

**Comment by vitaliihonta** [comment]
@pepegar Could you allow me to create branches? I'm done with the issue but cannot push

**Comment by pepegar** [comment]
I think is better if you fork the repo, push to your fork, and then create the PR from your fork to the main one.

**Comment by pepegar** [comment]
Closed in #99
